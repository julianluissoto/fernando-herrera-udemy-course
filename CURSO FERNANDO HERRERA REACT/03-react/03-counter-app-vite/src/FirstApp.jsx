import React from "react";
import PropTypes from "prop-types";
/* const funtionalValue = () => {
  return "value from function";
}; */
/* const valueObject = {
  name: "julian",
  lastname: "soto",
}; */

const FirstApp = ({ title, subtitle, number, name }) => {
  return (
    <>
      <h1 data-testid ='test-title'>{title}</h1>
      {/* <code>{JSON.stringify(valueObject)}</code> */}
      {/*  <h2>{funtionalValue()}</h2> */}
      <h2>{subtitle}</h2>
     
    </>
  );
};

export default FirstApp;

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  number: PropTypes.number,
};

FirstApp.defaultProps= {
title:"name by default props" , //if title is not pass by props it will use default prop
name: "julian soto"
}