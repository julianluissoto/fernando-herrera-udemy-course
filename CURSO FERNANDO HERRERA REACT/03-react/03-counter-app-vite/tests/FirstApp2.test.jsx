import { render, screen } from '@testing-library/react'

import FirstApp from '../src/FirstApp';


describe('Pruebas en <FirstApp />', () => { 
    
   const title= 'hola soy goku'
   const subtitle='Principe Saijajin'
test('should match with the snapshot ', () => {
    
    const {container} = render(<FirstApp title={title} />)

    expect(container).toMatchSnapshot();
});


test('should show message "hola, Soy Goku "', () => {
    
     render(<FirstApp title={title} />)
expect(screen.getByText( title )).toBeTruthy();
//screen.debug()

});

test('should show the title on an h1 tag ', () => {
    render(<FirstApp title={title} />)

    expect(screen.getByRole('heading', {level:1}).innerHTML).toContain(title);


});
test('should show the subtitle send by props', () => { 
    render(<FirstApp title={title} subtitle={subtitle}/>)
    expect(screen.getAllByText( subtitle ).length).toBe(1);
 })

 })