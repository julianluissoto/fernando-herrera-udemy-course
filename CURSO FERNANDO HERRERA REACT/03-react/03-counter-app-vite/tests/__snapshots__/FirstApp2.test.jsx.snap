// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> should match with the snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy goku
  </h1>
  <h2 />
</div>
`;
